### ISSUE TITLE
On iOS, initPlayer appears to work ... but logs errors and does not return capVideoPlayerResult with result: true

### ISSUE THREAD

**Issue by jh-gh** [jepiqueau/capacitor-video-player#78]
According to the API Docs, initPlayer should return a promise that resolves with a capVideoPlayerResult object (see https://github.com/jepiqueau/capacitor-video-player/blob/master/docs/API.md#initplayer).  This object should contain.a boolean property "result" with true indicating success. 

This worked on iOS on plugin version 3.3.0-1 (and 2.4.7).

However, with the plugin versions 3.4.1-1, 3.4.0 (and possibly any version between 3.3.0-1 and those versions) ... the promise resolves with an empty object.  The call actually appears to work, **but if apps check the result, they will think that it failed and act accordingly.**  

If you look at the logs in XCode (see below), it complains about being "Unable to activate audio session".  It also complains that it is unable to actually send the results back to the JS caller because it can't serialize the plugin response.

You can reproduce this problem using the Angular starter app https://github.com/jepiqueau/angular-videoplayer-app-starter

In the starter app, you won't actually see the problem directly as the video does appear to play correctly.  However, if you look at the logs (with Safari web inspector, or XCode) you will see the issue (where the app is receiving an empty object {} instead of the expected result object with result: true etc).  The starter app actually tries to display a toast with the error ... but this fails as well because it passes undefined for the message and Capacitor rejects the toast request.

Here is an example of the log from the starter app, running with plugin version 3.4.1-1 on iOS (captured from Safari).
<img width="589" alt="image" src="https://user-images.githubusercontent.com/4483416/156458635-46714e92-da80-4d74-820a-d95522ba9e5b.png">

By contrast, here is an example of the log from the same call in the starter app, running with plugin version 3.3.0.1 (captured from Safari)
<img width="567" alt="image" src="https://user-images.githubusercontent.com/4483416/156458763-062f71be-2651-4980-afb3-44bbf1c914a3.png">

Here is the more detailed log of the error in XCode (running with plugin version 3.4.1-1):
![image](https://user-images.githubusercontent.com/4483416/156458903-0d5ff435-aa1c-465a-9599-6a70df81e5af.png)

And here is the equivalent detailed log from XCode showing no error (running with plugin version 3.3.0-1):
<img width="661" alt="image" src="https://user-images.githubusercontent.com/4483416/156459049-519106e1-93a5-4d5b-8bad-6ba0dd700308.png">

**Comment by jepiqueau** [comment]
@jh-gh i cannot reproduce this error with `angular-videoplayer-app-starter` this is what i get in xcode when running the app
```
⚡️  TO JS {"value":true,"result":true,"method":"createVideoPlayerFullScreenView"}
⚡️  [log] - res {"value":true,"result":true,"method":"createVideoPlayerFullScreenView"}
⚡️  [log] - res.result  true
⚡️  TO JS {"currentTime":0,"videoRate":1,"fromPlayerId":"fullscreen"}
⚡️  TO JS {"currentTime":0,"fromPlayerId":"fullscreen","videoRate":1}
⚡️  TO JS {"videoRate":1,"currentTime":0,"fromPlayerId":"fullscreen"}
⚡️  [log] - Event jeepCapVideoPlayerReady [object Object]
⚡️  [log] - testVideoPlayerPlugin testAPI false
⚡️  [log] - testVideoPlayerPlugin first false
⚡️  [log] - Event jeepCapVideoPlayerPlay [object Object]
```

have you set in `Signing & Capabilities` the  `Background Modes` to allow `Audio, AirPlay, and Picture in Picture`
can you share the app on github and tell me in which simulator you test it

**Comment by jepiqueau** [comment]
@jh-gh to avoid the error on the toast do the following 
in line 61 of the `CapacitorVideoPlayerPlugin.Fullscreen.swit` file under `Extensions`
add `.localizedDescription` to `error` ie `"message": error.localizedDescription])` and you will see the message in the toast. This will be fixed in the next release. 
As i said, i do not know the reason why the audio session did not activate. This has no impact on the video playing except that when you will switch to another app the sound of the video will not continue to play in the background.

**Comment by jh-gh** [comment]
@jepiqueau Sorry for the delayed response.
I just tried reproducing the problem on a simulator (iphone 13, 15.2) and indeed I get the same results that you do (the call succeeds and returns the initPlayer call succeeds ... it does not hit an error trying to activate the audio stream ... and it does return the expected object containing result: true.   This is running with the current sources from https://github.com/jepiqueau/angular-videoplayer-app-starter.git (commit: a756572f23555633ecb426f671aa894a7fdf1606).  This is using 3.4.0 of the plugin.

Note that the above works fine (on the simulator) ... even though that starter project does NOT enable the background modes for audio/airplay etc.  It also works fine if I DO enable the background modes.

However, if I run the starter app on a real iphone ... I get the error "Unable to activate audio session" and the initPlayer() call resolves with an empty object (no result property).  If I add the background modes (audio, airplay, and pip) it makes no difference.  I still get the same error (again running on a physical device).

I have tried this on the following physical devices ... with and without the background modes ... and it fails in all cases:
- iphone 13 mini (15.3.1)
- iphone X (14.4.1)

Are you able to please try it on a physical iPhone and see if you can reproduce it that way?

Also, I think the reason for the secondary failure (where it just returns an empty object to the TS caller) is because the catch block that resolves the call after printing "Unable to activate audio session" tries to pass the whole error object back to Capacitor (instead of e.g. error.localizedDescription).

Thanks.

**Comment by jepiqueau** [comment]
@jh-gh read but i was waiting your feedback

**Comment by jepiqueau** [comment]
@jh-gh Can you try replacing lines 48,51 of CapacitorVideoPlayerPlugin.Fullscreen extension file
```
                        try self?.audioSession?
                            .setCategory(AVAudioSession.Category.playback,
                                         mode: AVAudioSession.Mode.moviePlayback,
                                         options: [.mixWithOthers, .allowAirPlay])

```
with 
```
                        try self?.audioSession?
                            .setCategory(playback,
                                         mode: .default,
                                         options: [.mixWithOthers])
```
and tell me if it works
Do you have any other Apps playing songs on your device when starting running the videoplayer app?

**Comment by jh-gh** [comment]
@jepiqueau Yes, sorry, forgot about your earlier comment where you pointed out the workaround for passing error.localizedDescription!

And yes, your fix above works on a physical iPhone!

I have tried 1) starting video with a song playing in Apple Music 2) starting video with nothing playing.  Both work.  In the case where there is a song playing ... it continues to play (and you hear both the song and the audio from the video).

Let me know if there are specific scenarios you would like me to test and thanks heaps for all the support.

**Comment by jepiqueau** [comment]
@jh-gh I am happy that it works i will modify the code for the two issues. Thanks for your proposal i will ask you for support when need it as i do not have a physical Apple device. I have four plugins under my ownership (`@capacitor-community/sqlite` , `@capacitor-community/photoviewer`, `capacitor-video-player`, `capacitor-data-storage-sqlite`) i do not know if you are using one of them if yes if you have any troubles when running them on a physical Apple device keep me aware it will help the community of developer.

**Comment by jh-gh** [comment]
@jepiqueau Just to confirm, have tested this fix in plugin version 3.4.1-3 on a physical iPhone and it works great, thanks.
